packages/shared/pkg/storage/storage_cache.go

```diff
@@ -204,6 +169,49 @@ var (
 	ErrBufferTooLarge  = errors.New("buffer is too large")
 )
 
+func (c *CachedFileObjectProvider) Size(ctx context.Context) (int64, error) {
+	if size, ok := c.readLocalSize(); ok {
+		cacheHits.Add(ctx, 1)
+		return size, nil
+	}
+	cacheMisses.Add(ctx, 1)
+
+	size, err := c.inner.Size(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	go c.writeLocalSize(size)
+
+	return size, nil
+}
+
+func (c *CachedFileObjectProvider) Delete(ctx context.Context) error {
+	return c.inner.Delete(ctx)
+}
+
+func (c *CachedFileObjectProvider) readLocalSize() (int64, bool) {
+	fname := c.sizeFilename()
+	content, err := os.ReadFile(fname)
+	if err != nil {
+		zap.L().Warn("failed to read cached size, falling back to remote read",
+			zap.String("path", fname),
+			zap.Error(err))
+		return 0, false
+	}
+
+	size, err := strconv.ParseInt(string(content), 10, 64)
+	if err != nil {
+		zap.L().Warn("failed to parse cached size, falling back to remote read",
```

Comment:
this shouldn't happen right? it probably should an error

Reply:
IMHO we use warnings when something bad happens that the program can recover gracefully from, and an error is when the program cannot recover. In this case, failing to read the cached size is fine b/c we can fall back to the remote size. If you feel strongly though, I can change it.

Reply:
Not sure if we have any framework for that, my heuristics here: 
- errors as some unexpected state
- warnings are expected, but they shouldn't happen by default